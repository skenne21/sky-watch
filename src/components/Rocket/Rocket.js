import React from 'react';
import PropTypes from 'prop-types';

const Rocket = ({ rocket, addRockets }) => {
  const { 
    name,
    summary,
    active,
    costLunch,
    fristFight
  } = rocket;
  
  const tags = Object.keys(rocket.stats).map( (stat, index) => {
    return <p key={stat+index}>{rocket.stats[stat]}</p>;
  });

  return (
    <div className='Rocket'>
      <h1>{name}</h1>
      <p>{active}</p>
      <h2>{costLunch}</h2>
      <h3>{fristFight}</h3>
      {tags}
      <p>{summary}</p>
      <button onClick={() => addRockets(rocket)}>X</button>
    </div>
  );
};

Rocket.propTypes = {
  rocket: PropTypes.object
};

export default Rocket;