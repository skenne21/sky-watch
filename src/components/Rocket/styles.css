.Rocket-card {
  display: inline-block;
  vertical-align: top;
  width: 25%;
  height: 680px;
  margin: 60px 10px;
  background-color: #ffffffa6;
}

.Rocket-card p {
  width: 90%;
  margin: 10px auto;
}

.Rocket-card_h1 {
  font-size: 1.8em;
  color: #1F3926;
}

.button {
  vertical-align: middle;
  background-image: url('../../assets/space-icon.png');
  background-size: cover;
  background-position: center;
  border-radius: 50%;
  width: 100px;
  height: 100px;
}