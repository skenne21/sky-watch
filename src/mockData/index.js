export const rawVideo = {
  items: [{
    id:{
      kind: "youtube#video", videoId: "XcVpMJp9Th4"
    },
    snippet: {
      channelId:"UC3OhT2ErtjLhvvkULJCCBlw",
      description:"Elon Musk gives new details, about SpaceX Mission...",
      publishedAt:"2017-09-29T06:05:03.000Z",
      thumbnails: {
        high: {
          url:"https://i.ytimg.com/vi/XcVpMJp9Th4/hqdefault.jpg",
          width: 480,
          height: 360
        }, 
        medium: {
          url:"https://i.ytimg.com/vi/XcVpMJp9Th4/mqdefault.jpg",
          width: 320,
          height: 360

        }
      },
      title:"Elon Musk"
    }
  }]
};

export const video = {
  id: "XcVpMJp9Th4",
  title: "Elon Musk",
  description: "Elon Musk gives new details, about SpaceX Mission...",
  date: "2017-09-29T06:05:03.000Z",
  image:{
    url: "https://i.ytimg.com/vi/XcVpMJp9Th4/hqdefault.jpg",
    width: 480,
    height: 360
  } 
};


export const videos = [{
  id: "XcVpMJp9Th4",
  title: "Elon Musk",
  description: "Elon Musk gives new details, about SpaceX Mission...",
  date: "2017-09-29T06:05:03.000Z",
  image:{
    url: "https://i.ytimg.com/vi/XcVpMJp9Th4/hqdefault.jpg",
    width: 480,
    height: 360
  } 
}, {
  id: "0PWu3BRxn60",
  title: "ZUMA Mission",
  description: "SpaceX is targeting launch...",
  date: "2018-01-08T01:19:10.000Z", 
  image: {
    url: "https://i.ytimg.com/vi/0PWu3BRxn60/hqdefault.jpg",
    width: 480,
    height: 360
  } 
}];

export const rawBio = {
  ceo:"Elon Musk",
  coo:"Gwynne Shotwell",
  cto:"Elon Musk",
  cto_propulsion:"Tom Mueller",
  employees:7000,
  founded:2002,
  founder:"Elon Musk",
  headquarters: {
    address: "Rocket Road",
    city: "Hawthorne",
    state: "California"
  },
  launch_sites:3,
  name:"SpaceX",
  summary:"SpaceX designs, manufactures and launches advanced...",
  test_sites:1,
  valuation:15000000000,
  vehicles:3
};

export const cleanBip = {
  name: "SpaceX",
  founder: "Elon Musk",
  founded: 2002,
  summary: "SpaceX designs, manufactures and launches advanced...",
  headquarters:{
    address: "Rocket Road",
    city: "Hawthorne",
    state: "California"
  }
};

export const bio = {
  name: "SpaceX",
  founder: "Elon Musk",
  founded: 2002,
  summary: "SpaceX designs, manufactures and launches advanced...",
  headquarters:{
    address: "Rocket Road",
    city: "Hawthorne",
    state: "California"
  }
};

export const rawRockets = [{
  active:false,
  first_flight: '2006-03-24',
  name:"Falcon 1",
  boosters:0,
  company:"SpaceX",
  cost_per_launch:6700000,
  country:"Republic of the Marshall Islands",
  description:"The Falcon 1 was an expendable launch system privately",
  diameter: {
    feet:5.5
  },
  engines: {
    engine_loss_max:0,
    layout:"single",
    number:1,
    propellant_1:"liquid oxygen",
    propellant_2:"RP-1 kerosene",
    thrust_sea_level: {
      kN:420,
      lbf:94000
    },
    thrust_vacuum: {
      kN:480,
      lbf:110000
    },
    type:"merlin"
  },
  height: {
    feet:73
  },
  mass: {
    lb:66460
  },
  payload_weights: [{id:1}]
}, {
  active:true,
  first_flight: '2006-03-24',
  name:"Falcon 9",
  boosters:0,
  company:"SpaceX",
  cost_per_launch:6700000,
  country:"United States",
  description:"The Falcon 1 was an expendable launch system privately",
  diameter: {
    feet:5.5
  },
  engines: {
    engine_loss_max:0,
    layout:"single",
    number:1,
    propellant_1:"liquid oxygen",
    propellant_2:"RP-1 kerosene",
    thrust_sea_level: {
      kN:420,
      lbf:94000
    },
    thrust_vacuum: {
      kN:480,
      lbf:110000
    },
    type:"merlin"
  },
  height: {
    feet:73
  },
  mass: {
    lb:1207920
  },
  payload_weights: [{id:1}]
}];

export const cleanRocket = {
  active:false,
  costLaunch:6700000,
  country:"Republic of the Marshall Islands",
  fristFight:"03, 24, 2006",
  name:"Falcon 1",
  stats: {
    boosters:0,
    diameter:5.5,
    height:73,
    mass:66460
  },
  summary:"The Falcon 1 was an expendable ..."
};

export const rockets =  [
  {
  "name": "Name: Falcon 1", 
    "stats": {
      "active": "Active: false",
      "boosters": "Boosters: 0",
      "costLaunch": "Cost To Launch:6700000", 
      "country": "Launch From: Republic of the Marshall Islands", 
      "description": "Description: The Falcon 1 was an expendable launch system privately", 
      "diameter": "Diameter 5.5 ft", "fristFlight": "Frist Flight: 03, 24, 2006", "height": 
      "Height: 73 ft", 
      "mass": "Mass: 66460 lb"
    }, "type": "rocket"
  }, {
    "name": "Name: Falcon 9", 
    "stats": {
      "active": "Active: true",
      "boosters": "Boosters: 0",
      "costLaunch": "Cost To Launch:6700000", 
      "country": "Launch From: United States", 
      "description": "Description: The Falcon 1 was an expendable launch system privately", 
      "diameter": "Diameter 5.5 ft",
      "fristFlight": "Frist Flight: 03, 24, 2006", 
      "height": "Height: 73 ft", 
      "mass": "Mass: 1207920 lb"
    }, 
    "type": "rocket"
  }
];

export const rawCapsules = [
  {
    active:true,
    crew_capacity:0,
    diameter: {
      feet:12,
      meters:3.7
    },
    heat_shield: {
      material:"PICA-X",
      size_meters:3.6,
      temp_degrees:3000
    },
    height_w_trunk: {
      feet:23.6,
      meters:7.2
    },
    name:"Dragon 1",
    orbit_duration_yr: 2,
    launch_payload_mass:  {
      kg: 3000,
      lb: 6614
    },
  }, {
    active:true,
    crew_capacity:0,
    diameter: {
      feet:12,
      meters:3.7
    },
    heat_shield: {
      material:"PICA-X",
      size_meters:3.6,
      temp_degrees:3000
    },
    height_w_trunk: {
      feet:23.6,
      meters:7.2
    },
    name:"Dragon 1",
    orbit_duration_yr: 2,
    launch_payload_mass:  {
      kg: 3000,
      lb: 6614
    },
  }
];

export const capsules = [
  {
    "name": "Name: Dragon 1",
    "stats": {
      "crew": "Crew Capacity: 0", 
      "diameter": "Diameter: 12 ft", 
      "heatShield": "Heat Shield Capacity: 3000 deg", 
      "height": "Height: 23.6 ft", 
      "mass": "Mass: 3000 kg", 
      "yearInOrbit": " Years in Orbit: 2"
    }, 
    "type": "capsule"
  }, {
    "name": "Name: Dragon 1", 
    "stats": {
      "crew": "Crew Capacity: 0", 
      "diameter": "Diameter: 12 ft", 
      "heatShield": "Heat Shield Capacity: 3000 deg", 
      "height": "Height: 23.6 ft", 
      "mass": "Mass: 3000 kg", 
      "yearInOrbit": " Years in Orbit: 2"
    }, 
    "type": "capsule"
  }
];

export const capsule = {
  "name": "Name: Dragon 1", 
  "stats": {
    "crew": "Crew Capacity: 0", 
    "diameter": "Diameter: 12 ft", 
    "heatShield": "Heat Shield Capacity: 3000 deg", 
    "height": "Height: 23.6 ft", 
    "mass": "Mass: 3000 kg", 
    "yearInOrbit": " Years in Orbit: 2"
  }, 
  "type": "capsule"
};

export const rawLaunchpads = [
  {
    details:"SpaceX original launch site...",
    full_name:"Kwajalein Atoll Omelek Island",
    id:"kwajalein_atoll",
    location: {
      latitude:9.0477206,
      longitude:167.7431292,
      name:"Omelek Island",
      region:"Marshall Islands"
    },
    status:"retired",
    vehicles_launched: ['Falcon 1']
  }, {
    details:"SpaceX east coast landing...",
    full_name:"Cape Canaveral Air Force Station, Space Launch Complex 13",
    id:"kwajalein_atoll",
    location: {
      latitude:9.0477206,
      longitude:167.7431292,
      name:"Cape Canaveral",
      region:"Florida"
    },
    status:"active",
    vehicles_launched: ['Falcon 9']
  }
];

export const launchpads = [
  {
    "name": "Name: Kwajalein Atoll Omelek Island", 
    "stats": {
      "details": "SpaceX original launch site...", 
      "location": "Location: Marshall Islands,Omelek Island", 
      "status": "Status: retired", 
      "vehicles": "Vehicles Launched: Falcon 1"
    }, 
    "type": "launchpad"
  }, {
    "name": "Name: Cape Canaveral Air Force Station, Space Launch Complex 13", 
    "stats": {
      "details": "SpaceX east coast landing...",
      "location": "Location: Florida,Cape Canaveral", 
      "status": "Status: active", 
      "vehicles": "Vehicles Launched: Falcon 9"
    }, 
    "type": "launchpad"
  }
];

export const launchpad = {
  details:"SpaceX east coast landing...", 
  location:"Florida: Cape Canaveral",
  name:"Cape Canaveral Air Force Station, Space Launch Complex 13",
  status:"active",
  vehicles:["Falcon 9"]
}

export const launches = {
  falcon1:[
  {
    name: "Flight Number: 1",
    type: "launches",
    image: "https://www.spacexnow.com/patches/falcon1-1.png",
    video: "https://www.youtube.com/watch?v=0a_00nJ_Y88",
    stats : {
      site: "Kwajalein Atoll",
      sucessful: "Launch Sucess: false",
      details: "Details: Engine failure at 33 seconds and loss of vehicle",
      capsule: "Reuseable Capsule: false",
      core: "Reuseable Core: false",
      sideCores: "Reusable Sidecores: false",
      fairings: "Reusable Fairings: false",
    } 
  }, {
    name: "Flight Number: 2",
    type: "launches",
    image: "https://www.spacexnow.com/patches/falcon1-1.png",
    video: "https://www.youtube.com/watch?v=0a_00nJ_Y88",
    stats : {
      site: "Kwajalein Atoll",
      sucessful: "Launch Sucess: false",
      details: "Details: Engine failure at 33 seconds and loss of vehicle",
      capsule: "Reuseable Capsule: false",
      core: "Reuseable Core: false",
      sideCores: "Reusable Sidecores: false",
      fairings: "Reusable Fairings: false",
    }
  }],
  falcon9:[
  {
    name: "Flight Number: 1",
    type: "launches",
    image: "https://www.spacexnow.com/patches/falcon1-1.png",
    video: "https://www.youtube.com/watch?v=0a_00nJ_Y88",
    stats : {
      site: "Kwajalein Atoll",
      sucessful: "Launch Sucess: false",
      details: "Details: Engine failure at 33 seconds and loss of vehicle",
      capsule: "Reuseable Capsule: false",
      core: "Reuseable Core: false",
      sideCores: "Reusable Sidecores: false",
      fairings: "Reusable Fairings: false",
    } 
  }, {
    name: "Flight Number: 2",
    type: "launches",
    image: "https://www.spacexnow.com/patches/falcon1-1.png",
    video: "https://www.youtube.com/watch?v=0a_00nJ_Y88",
    stats : {
      site: "Kwajalein Atoll",
      sucessful: "Launch Sucess: false",
      details: "Details: Engine failure at 33 seconds and loss of vehicle",
      capsule: "Reuseable Capsule: false",
      core: "Reuseable Core: false",
      sideCores: "Reusable Sidecores: false",
      fairings: "Reusable Fairings: false",
    }
  }],
  falconHeavy: [
  {
    name: "Flight Number: 2",
    type: "launches",
    image: "https://www.spacexnow.com/patches/falcon1-1.png",
    video: "https://www.youtube.com/watch?v=0a_00nJ_Y88",
    stats : {
      site: "Kwajalein Atoll",
      sucessful: "Launch Sucess: false",
      details: "Details: Engine failure at 33 seconds and loss of vehicle",
      capsule: "Reuseable Capsule: false",
      core: "Reuseable Core: false",
      sideCores: "Reusable Sidecores: false",
      fairings: "Reusable Fairings: false",
    }
  }] 
}
