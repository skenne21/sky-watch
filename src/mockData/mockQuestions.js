export const questions = [
  {
    name: '',
    type: 'questionCard',
    question: 'Why Did CEO Elon Musk name the spacecraft Dragon?',
    stats: {
      answer: `Elon Musk has said that he named his spacecraft "Dragon"
       after the fictional Puff the Magic Dragon, from the hit song by 
       music group Peter, Paul and Mary. Musk said he used the name 
       because many critics considered his goals impossible when he 
       founded SpaceX in 2002.`
    }
  }, {
    name: '',
    type: 'questionCard',
    question: `What was carried on the Dragon capsule's first test 
    flight in December 2010?`,
    stats: {
      answer: `It carried a wheel of cheese into orbit. Prior to the 
      successful liftoff, SpaceX officials hinted that Dragon was carrying 
      special cargo into space, but the company did not reveal the secret 
      payload until after the spacecraft had returned.
        
      The wheel of cheese was launched in honor of a classic skit from 
      actor John Cleese in the British comedy show Monty Python's
      Flying Circus.

      The cheese rode inside the Dragon capsule for the entire flight, from the 
      launch in Florida to splashdown in the Pacific Ocean, 500 miles 
      (804 kilometers) west of Mexico.`
    }
  }, {
    name: '',
    type: 'questionCard',
    question: `What Are SpaceX's fleet of Falcon rockets named affter?`,
    stats: {
      answer: `the Falcon 9 boosters and smaller Falcon 1 rockets are 
      named after the fictional Millenium Falcon spaceship from creator 
      George Lucas' "Star Wars" movies.`
    }
  }, {
    name: '',
    type: 'questionCard',
    question: `What Marvel character is CEO Elon Musk said to be a portrayal 
    of?`,
    stats: {
      answer: `Musk, who founded SpaceX in 2002, is also the CEO of Tesla 
      Motors
       and made his initial fortune by co-founding the online payment system 
       PayPal. 
      The entrepreneur has also developed a larger-than-life persona, with 
      admirers
       calling him the real-life Tony Stark`
    }
  }, {
    name: '',
    type: 'questionCard',
    question: `What is the metophor SpaceX has said to explain the 
    need for reuseable spacecrafts `,
    stats: {
      answer: `A metaphor Elon would frequently use with us is 
      comparing it to throwing away a Boeing 747 after every 
      flight. We simply can't go forward in space exploration if 
      the vehicles don't become reusable.`
    }
  }, {
    name: '',
    type: 'questionCard',
    question: `What was the reason for Falcon 9 first-stage landing test `,
    stats: {
      answer: `Falcon 9 first-stage landing test were a series of 
      controlled-descent flight tests conducted by SpaceX between 2013 
      and 2016. Since 2017, the first stage of Falcon 9 missions has
      been routinely landed if the rocket performance allowed it, 
      and if SpaceX choose to recover the stage.

      The program's objective was to reliably execute controlled re-entry,
      descent and landing (EDL) of the Falcon 9 first stage into Earth's 
      atmosphere after the stage completes the boost phase of an orbital
      spaceflight. The first tests aimed to touch down vertically in the
      ocean at zero velocity. Later tests attempted to land the rocket
      precisely on an autonomous spaceport drone ship (a barge 
      commissioned by SpaceX to provide a stable landing surface at sea)
      or at Landing Zone 1 (LZ-1), a concrete pad at Cape Canaveral.
      The first ground landing at LZ-1 succeeded in December 2015,
      and the first landing at sea on a drone ship in April 2016.
      The second landed booster, B1021, was the first to fly again
      in March 2017, and was recovered a second time.`
    }
  }, {
    name: '',
    type: 'questionCard',
    question: `From 2013 to 2016, sixteen test flights were conducted
    Falcon 9 first-stage landing test how many were successful?`,
    stats: {
      answer: `From 2013 to 2016, sixteen test flights were conducted,
      six of which achieved a soft landing and recovery of the booster.`
    }
  }, {
    name: '',
    type: 'questionCard',
    question: `What Falcon 9 achieving a stable landing at sea?`,
    stats: {
      answer: `Falcon 9 flight 23 (CRS-8) finally achieved a stable
      landing at sea in the Atlantic on the drone ship Of Course I Still
      Love You in April 2016 after four previous attempts ended in
      destruction of the booster upon impact`
    }
  }
];

export const randomQuestions = () => {
  const randomIndex = Math.floor(Math.random() * (8- 1)) + 1;
  const card = questions.filter((question, index) => index === randomIndex);
  const newArray = [];
  newArray.push(card[0]);
  return newArray;
};