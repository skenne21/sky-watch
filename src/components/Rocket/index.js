import React from 'react';
import PropTypes from 'prop-types';
import './styles.css';

const Rocket = ({ rocket, addRockets }) => {
  const { 
    name,
    summary,
    active,
    costLaunch,
    fristFight
  } = rocket;
  
  const tags = Object.keys(rocket.stats).map( (stat, index) => {
    return <p key={stat+index}>{rocket.stats[stat]}</p>;
  });

  return (
    <div className='Rocket-card'>
      <h1>Name: {name}</h1>
      <p>Active: {active}</p>
      <h2>Cost to Launch {costLaunch}</h2>
      <h3>Date of Frist Fight : {fristFight}</h3>
      {tags}
      <p>{summary}</p>
      <button
        className='button'
        onClick={() => addRockets(rocket)}>
      </button>
    </div>
  );
};

Rocket.propTypes = {
  rocket: PropTypes.object
};

export default Rocket;