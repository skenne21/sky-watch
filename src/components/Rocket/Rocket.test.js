import React from 'react';
import { shallow } from 'enzyme';
import Rocket from './Rocket';
import * as mocks from '../../mockData';

describe('Rocket', () =>{
  it("Should match the snapshot", () => {
    const rocket = mocks.cleanRocket;
    const wrapper = shallow(<Rocket rocket={rocket}/>);
    expect(wrapper).toMatchSnapshot();
  });
});
