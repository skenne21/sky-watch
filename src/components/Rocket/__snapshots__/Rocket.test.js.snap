// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rocket Should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Rocket
    rocket={
        Object {
            "active": false,
            "costLaunch": 6700000,
            "country": "Republic of the Marshall Islands",
            "fristFight": "03, 24, 2006",
            "name": "Falcon 1",
            "stats": Object {
              "boosters": 0,
              "diameter": 5.5,
              "height": 73,
              "mass": 66460,
            },
            "summary": "The Falcon 1 was an expendable ...",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1>
          Falcon 1
</h1>,
        <p />,
        <h2 />,
        <h3>
          03, 24, 2006
</h3>,
        Array [
          <p />,
          <p>
            5.5
</p>,
          <p>
            73
</p>,
          <p>
            66460
</p>,
        ],
        <p>
          The Falcon 1 was an expendable ...
</p>,
      ],
      "className": "Rocket",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Falcon 1",
        },
        "ref": null,
        "rendered": "Falcon 1",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": false,
        },
        "ref": null,
        "rendered": false,
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "03, 24, 2006",
        },
        "ref": null,
        "rendered": "03, 24, 2006",
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": "boosters0",
        "nodeType": "host",
        "props": Object {
          "children": 0,
        },
        "ref": null,
        "rendered": 0,
        "type": "p",
      },
      Object {
        "instance": null,
        "key": "diameter1",
        "nodeType": "host",
        "props": Object {
          "children": 5.5,
        },
        "ref": null,
        "rendered": 5.5,
        "type": "p",
      },
      Object {
        "instance": null,
        "key": "height2",
        "nodeType": "host",
        "props": Object {
          "children": 73,
        },
        "ref": null,
        "rendered": 73,
        "type": "p",
      },
      Object {
        "instance": null,
        "key": "mass3",
        "nodeType": "host",
        "props": Object {
          "children": 66460,
        },
        "ref": null,
        "rendered": 66460,
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "The Falcon 1 was an expendable ...",
        },
        "ref": null,
        "rendered": "The Falcon 1 was an expendable ...",
        "type": "p",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1>
            Falcon 1
</h1>,
          <p />,
          <h2 />,
          <h3>
            03, 24, 2006
</h3>,
          Array [
            <p />,
            <p>
              5.5
</p>,
            <p>
              73
</p>,
            <p>
              66460
</p>,
          ],
          <p>
            The Falcon 1 was an expendable ...
</p>,
        ],
        "className": "Rocket",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Falcon 1",
          },
          "ref": null,
          "rendered": "Falcon 1",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": false,
          },
          "ref": null,
          "rendered": false,
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "03, 24, 2006",
          },
          "ref": null,
          "rendered": "03, 24, 2006",
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": "boosters0",
          "nodeType": "host",
          "props": Object {
            "children": 0,
          },
          "ref": null,
          "rendered": 0,
          "type": "p",
        },
        Object {
          "instance": null,
          "key": "diameter1",
          "nodeType": "host",
          "props": Object {
            "children": 5.5,
          },
          "ref": null,
          "rendered": 5.5,
          "type": "p",
        },
        Object {
          "instance": null,
          "key": "height2",
          "nodeType": "host",
          "props": Object {
            "children": 73,
          },
          "ref": null,
          "rendered": 73,
          "type": "p",
        },
        Object {
          "instance": null,
          "key": "mass3",
          "nodeType": "host",
          "props": Object {
            "children": 66460,
          },
          "ref": null,
          "rendered": 66460,
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "The Falcon 1 was an expendable ...",
          },
          "ref": null,
          "rendered": "The Falcon 1 was an expendable ...",
          "type": "p",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
